### ISSUE TITLE
Game crashes when writing on parchment

### ISSUE THREAD

**Issue by SheriffEllis** [anegostudios/VintageStory-Issues#2629]
### Game Version

1.18.0

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

When writing on parchment with ink and quill the game will crash due to "Index out of range" exception when arrow key is held to move cursor along text.

### How to reproduce

1. Hold parchment with ink and quill in offhand.
2. Write some text into the parchment.
3. Attempt to move the text cursor along the text using only the arrow keys (it will not happen immediately so hold it for a while).

### Logs

```Running on 64 bit Windows with 16 GB RAM 
Game Version: v1.18.0 (Stable)
21/04/2023 01:44:55: Critical error occurred
Loaded Mods: bettercrates@1.6.0, foundrymod@1.0.0, qptech@1.13.01, rainfrogs@1.0.0, roas@1.0.0, rustandrails@1.0.4, game@1.18.0, alloycalculator@1.0.0, animalcages@2.1.4, betterscything@0.1.0, carryon@1.0.0-rc.3, commonlib@2.0.4, driftersounds@1.0.0, hudclock@2.9.1, medievalexpansion@3.9.0, prospectorinfo@4.1.4, techmolds@1.0.0, useplanksinpitkiln@1.0.0, creative@1.18.0, survival@1.18.0, workbenchexpansion@1.6.0, xlib@0.7.0-pre.3, chemistrylib@1.1.0, treetapping@1.0.0, xinvtweaks@1.5.1, xskills@0.7.0-pre.5, kosphotography@1.1.2
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at Vintagestory.GameContent.GuiDialogEditableBook.onTryTextChange(List`1 lines) in VSSurvivalMod\Systems\WritingSystem\GuiDialogEditableBook.cs:line 78
   at Vintagestory.API.Client.GuiElementEditableTextBase.LoadValue(List`1 newLines) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Text\GuiElementEditableTextBase.cs:line 281
   at Vintagestory.API.Client.GuiElementEditableTextBase.SetValue(String text, Boolean setCaretPosToEnd) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Text\GuiElementEditableTextBase.cs:line 266
   at Vintagestory.GameContent.GuiDialogReadonlyBook.updatePage(Boolean setCaretPosToEnd) in VSSurvivalMod\Systems\WritingSystem\GuiDialogReadonlyBook.cs:line 174
   at Vintagestory.GameContent.GuiDialogEditableBook.OnKeyDown(KeyEvent args) in VSSurvivalMod\Systems\WritingSystem\GuiDialogEditableBook.cs:line 239
   at Vintagestory.Client.NoObf.GuiManager.OnKeyDown(KeyEvent args) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 361
   at Vintagestory.Client.NoObf.ClientMain.OnKeyDown(KeyEvent args) in VintagestoryLib\Client\ClientMain.cs:line 1736
   at Vintagestory.Client.NoObf.ClientPlatformWindows.game_KeyDown(Object sender, KeyboardKeyEventArgs e) in VintagestoryLib\Client\ClientPlatform\Input.cs:line 250
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.Platform.NativeWindowBase.OnKeyDown(Key key, Boolean repeat) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\Platform\NativeWindowBase.cs:line 130
   at OpenTK.Platform.Windows.WinGLNative.HandleKeyboard(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 646
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 828
   at OpenTK.Platform.Windows.Functions.DispatchMessage(MSG& msg)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents() in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 1551
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 369
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs)
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 04/04/2023 01:36:15, Site = , Source = Windows Error Reporting, Message = Fault bucket 2067791412495284563, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.18.0.0
P3: 10.0.19044.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\sheri\AppData\Local\Temp\RDR7414.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7424.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7435.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7443.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7473.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: d9d4ac53-5642-474e-98cc-6292127f572b
Report Status: 268435456
Hashed bucket: f60bf9350d4e0c5a0cb2455583627953
Cab Guid: 0 }
--------------
{ TimeGenerated = 11/01/2023 12:36:11, Site = , Source = Windows Error Reporting, Message = Fault bucket 2054915892851142415, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.17.10.0
P3: 10.0.19044.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\sheri\AppData\Local\Temp\RDRF531.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF541.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF571.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF58E.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERF5CE.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: b39615f2-cd80-4cb8-b195-8db69a502552
Report Status: 268435456
Hashed bucket: 25c2d65fe412a5f0ec84871dfcb26f0f
Cab Guid: 0 }```

**Comment by radfast** [comment]
Duplicate of #2572, but for parchment.  For now let's keep both open for visibility

**Comment by radfast** [comment]
Fixed in upcoming 1.18.2 stable, thank you for reporting
